pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -160,9 +148,16 @@ func (m *manager) shouldRotate() bool {
 	m.certAccessLock.RLock()
 	defer m.certAccessLock.RUnlock()
 	notAfter := m.cert.Leaf.NotAfter
-	total := notAfter.Sub(m.cert.Leaf.NotBefore)
-	remaining := notAfter.Sub(time.Now())
-	return remaining < 0 || uint(remaining*100/total) < m.shouldRotatePercent
+	totalDuration := float64(notAfter.Sub(m.cert.Leaf.NotBefore))
+
+	// Use some jitter to set the rotation threshold so each node will rotate
+	// at approximately 70-90% of the total lifetime of the certificate.  With
+	// jitter, if a number of nodes are added to a cluster at approximately the
+	// same time (such as cluster creation time), they won't all try to rotate
+	// certificates at the same time for the rest of the life of the cluster.
+	jitteryDuration := wait.Jitter(time.Duration(totalDuration*0.2), 1.0) + time.Duration(totalDuration*0.5)
```

Comment:
I think 0.5 is correct. My reasoning is, [Jitter documentation](https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Jitter):

    Jitter returns a time.Duration between duration and duration + maxFactor * duration.

So it returns a value between the first parameter (0.2) and the first parameter + (first parameter * second parameter). So, the result of Jitter should be in the range of [0.2, 0.2 + (0.2 * 1)] = [0.2, 0.4].